internal/pkg/cli/pipeline_init.go

```diff
@@ -355,24 +355,51 @@ func relPath(fullPath string) string {
 	return relPath
 }
 
-// TODO: Nice-to-have: have an opts.listRemoteRepos() method that execs out to `git remote -v` and parse repo name to offer select menu
-func (opts *InitPipelineOpts) selectGitHubRepo() error {
-	repo, err := opts.prompt.Get(
-		pipelineEnterGitHubRepoPrompt,
-		fmt.Sprintf(`The GitHub repository linked to your workspace. Pushing to this repository will trigger your pipeline build stage. Please enter full repository URL, e.g. "https://github.com/myCompany/myRepo", or the owner/rep, e.g. "myCompany/myRepo"`),
-		validateGitHubRepo,
+func (opts *InitPipelineOpts) selectGitHubOwner() error {
+	owner, err := opts.prompt.SelectOne(
+		pipelineSelectGitHubOwnerPrompt,
+		pipelineSelectGitHubOwnerHelpPrompt,
+		opts.owners,
 	)
-
-	// Remove ".git" at the end of the Github repo URL.
-	repoNameSplit := strings.Split(repo, ".")
-	if repoNameSplit[len(repoNameSplit)-1] == "git" {
-		repo = strings.Join(repoNameSplit[:len(repoNameSplit)-1], ".")
+	if err != nil {
+		return fmt.Errorf("get GitHub owner name: %w", err)
 	}
+	opts.GitHubOwner = owner
 
+	return nil
+}
+
+func parseGitRemoteResult(s string) ([]string, []string) {
+	var owners, repos []string
+	ownerSet := make(map[string]bool)
+	repoSet := make(map[string]bool)
+	reOwner := regexp.MustCompile(`\:.*\/`)
+	reRepo := regexp.MustCompile(`\/.*\.git`)
+	items := strings.Split(s, "\n")
+	for _, item := range items {
+		owner := strings.TrimSuffix(strings.TrimPrefix(reOwner.FindString(item), ":"), "/")
+		repo := strings.TrimSuffix(strings.TrimPrefix(reRepo.FindString(item), "/"), ".git")
```

Comment:
We can have all sorts of weird remote URIs unfortunately 😔 
1. https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes (some end with .git, some don't)
2. https://stackoverflow.com/a/44670888/1201381 (owner can be difficult to get)

I believe these are the 4 options that you can get for GitHub remotes:
```
   efekarakus	git@github.com:efekarakus/grit.git (fetch)
   efekarakus	https://github.com/karakuse/grit.git (fetch)
   origin	https://github.com/koke/grit (fetch)
   koke      git://github.com/koke/grit.git (push)
```

Maybe we can update the regex to match these patterns?

Reply:
Regardless of whatever logic we choose, 
1. It'd be nice to provide the user a way of entering a custom value if we don't manage to detect the repo or owner with an option "Enter custom value"
2. We should do some more involved unit testing for the various values of `remote -v`.